options.become_user is None in AnsibleRunnerV2 class

When using ansible backend against remote hosts that require privilege escalation, the run fails because of 'None' value in self.options.become_user field of AnsibleRunnerV2 class.

The Ansible tool addresses this issue by running ansible.cli.CLI.normalize_become_options() method, which is not static and requires a CLI object.

```
(test-infra)realloc@sunset:~/projects/test-infra$ cat test/base/test_python.py 

def test_python27_is_installed(Package):
    python27 = Package("python2.7")
    assert python27.is_installed

(test-infra)realloc@sunset:~/projects/test-infra$ testinfra --connection=ansible -v test/base/ --hosts=ctrl
============================================================= test session starts ==============================================================
platform linux2 -- Python 2.7.6, pytest-3.0.5, py-1.4.32, pluggy-0.4.0 -- /home/realloc/projects/test-infra/venv/bin/python2
cachedir: .cache
rootdir: /home/realloc/projects/test-infra, inifile: 
plugins: testinfra-1.5.1
collected 1 items 

test/base/test_python.py::test_python27_is_installed[ansible://ctrl] ERROR

==================================================================== ERRORS ====================================================================
_________________________________________ ERROR at setup of test_python27_is_installed[ansible://ctrl] _________________________________________

TestinfraBackend = <testinfra.backend.ansible.AnsibleBackend object at 0x7fe5fd1f2ad0>

    @pytest.fixture()
    def f(TestinfraBackend):
>       return TestinfraBackend.get_module(cls.__name__)

venv/local/lib/python2.7/site-packages/testinfra/modules/base.py:71: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
venv/local/lib/python2.7/site-packages/testinfra/backend/base.py:250: in get_module
    module = getattr(testinfra.modules, name).get_module(self)
venv/local/lib/python2.7/site-packages/testinfra/modules/base.py:58: in get_module
    klass = cls.get_module_class(_backend)
venv/local/lib/python2.7/site-packages/testinfra/modules/package.py:68: in get_module_class
    if SystemInfo.type == "freebsd":
venv/local/lib/python2.7/site-packages/testinfra/modules/systeminfo.py:104: in type
    return self.sysinfo["type"]
venv/local/lib/python2.7/site-packages/testinfra/modules/systeminfo.py:31: in sysinfo
    self._sysinfo = self.get_system_info()
venv/local/lib/python2.7/site-packages/testinfra/modules/systeminfo.py:87: in get_system_info
    sysinfo["type"] = self.check_output("uname -s").lower()
venv/local/lib/python2.7/site-packages/testinfra/modules/base.py:23: in run
    return self._backend.run(command, *args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <testinfra.backend.ansible.AnsibleBackend object at 0x7fe5fd1f2ad0>, command = 'uname -s', args = ()
out = {'exception': 'Traceback (most recent call last):
  File "/home/realloc/projects/test-infra/venv/local/lib/python2.7/...oneType\' object is not iterable
', 'failed': True, 'msg': 'Unexpected failure during module execution.', 'stdout': ''}
stdout_bytes = ''

    def run(self, command, *args):
        command = self.get_command(command, *args)
        out = self.run_ansible("shell", module_args=command)
    
        # Ansible may return bytes as an unicode object...
        # A simple test case is:
        # >>> assert File("/bin/true").content == open("/bin/true").read()
    #        print out
        try:
            stdout_bytes = b"".join((chr(ord(c)) for c in out['stdout']))
        except ValueError:
            stdout_bytes = None
    
        try:
>           stderr_bytes = b"".join((chr(ord(c)) for c in out['stderr']))
E           KeyError: u'stderr'

venv/local/lib/python2.7/site-packages/testinfra/backend/ansible.py:56: KeyError
------------------------------------------------------------ Captured stdout setup -------------------------------------------------------------
{'subset': None, 'ask_pass': False, 'become_user': None, 'sudo': False, 'private_key_file': None, 'syntax': None, 'skip_tags': None, 'diff': False, 'sftp_extra_args': '', 'check': False, 'force_handlers': False, 'remote_user': 'sv.bogatyrev', 'become_method': 'sudo', 'vault_password_file': 'bin/vault-pass.sh', 'output_file': None, 'ask_su_pass': False, 'new_vault_password_file': None, 'inventory': 'env/vagrant/hosts', 'forks': 5, 'listhosts': None, 'ssh_extra_args': '', 'tags': 'all', 'become_ask_pass': False, 'flush_cache': None, 'become': True, 'su_user': None, 'ask_sudo_pass': False, 'extra_vars': [], 'verbosity': 0, 'su': False, 'ssh_common_args': '', 'connection': u'smart', 'ask_vault_pass': False, 'timeout': 10, 'module_path': None, 'sudo_user': None, 'scp_extra_args': ''}
RUN Ansible(u'shell',u'uname -s',{}):{'exception': u'Traceback (most recent call last):\n  File "/home/realloc/projects/test-infra/venv/local/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 123, in run\n    res = self._execute()\n  File "/home/realloc/projects/test-infra/venv/local/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 447, in _execute\n    result = self._handler.run(task_vars=variables)\n  File "/home/realloc/projects/test-infra/venv/local/lib/python2.7/site-packages/ansible/plugins/action/normal.py", line 33, in run\n    results = merge_hash(results, self._execute_module(tmp=tmp, task_vars=task_vars))\n  File "/home/realloc/projects/test-infra/venv/local/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 618, in _execute_module\n    self._fixup_perms(tmp, remote_user, recursive=True)\n  File "/home/realloc/projects/test-infra/venv/local/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 340, in _fixup_perms\n    res = self._remote_chown(remote_path, self._play_context.become_user, recursive=recursive)\n  File "/home/realloc/projects/test-infra/venv/local/lib/python2.7/site-packages/ansible/plugins/action/__init__.py", line 393, in _remote_chown\n    cmd = self._connection._shell.chown(path, user, group, recursive=recursive)\n  File "/home/realloc/projects/test-infra/venv/local/lib/python2.7/site-packages/ansible/plugins/shell/__init__.py", line 72, in chown\n    user = pipes.quote(user)\n  File "/usr/lib/python2.7/pipes.py", line 269, in quote\n    for c in file:\nTypeError: \'NoneType\' object is not iterable\n',
 'failed': True,
 'msg': 'Unexpected failure during module execution.',
 'stdout': ''}
============================================================ pytest-warning summary ============================================================
WP1 None Module already imported so can not be re-written: testinfra
================================================== 1 pytest-warnings, 1 error in 0.60 seconds ==================================================
```

--- a/testinfra/utils/ansible_runner.py
+++ b/testinfra/utils/ansible_runner.py
@@ -138,7 +138,8 @@ class AnsibleRunnerV2(AnsibleRunnerBase):
         super(AnsibleRunnerV2, self).__init__(host_list)
         _reload_constants()
         self.variable_manager = ansible.vars.VariableManager()
-        self.options = ansible.cli.CLI(None).base_parser(
+        self.cli = ansible.cli.CLI(None)
+        self.cli.options = ansible.cli.CLI(None).base_parser(
             connect_opts=True,
             meta_opts=True,
             runas_opts=True,
@@ -150,17 +151,18 @@ class AnsibleRunnerV2(AnsibleRunnerBase):
             fork_opts=True,
             module_opts=True,
         ).parse_args([])[0]
-        self.options.connection = "smart"
+        self.cli.normalize_become_options()
+        self.cli.options.connection = "smart"
         self.loader = ansible.parsing.dataloader.DataLoader()
-        if self.options.vault_password_file:
+        if self.cli.options.vault_password_file:
             vault_pass = ansible.cli.CLI.read_vault_password_file(
-                self.options.vault_password_file, loader=self.loader)
+                self.cli.options.vault_password_file, loader=self.loader)
             self.loader.set_vault_password(vault_pass)
 
         self.inventory = ansible.inventory.Inventory(
             loader=self.loader,
             variable_manager=self.variable_manager,
-            host_list=host_list or self.options.inventory,
+            host_list=host_list or self.cli.options.inventory,
         )
         self.variable_manager.set_inventory(self.inventory)
 
@@ -175,7 +177,7 @@ class AnsibleRunnerV2(AnsibleRunnerBase):
             self.loader, host=self.inventory.get_host(host))
 
     def run(self, host, module_name, module_args=None, **kwargs):
-        self.options.check = kwargs.get("check", False)
+        self.cli.options.check = kwargs.get("check", False)
         action = {"module": module_name}
         if module_args is not None:
             if module_name in ("command", "shell"):
@@ -196,7 +198,7 @@ class AnsibleRunnerV2(AnsibleRunnerBase):
                 inventory=self.inventory,
                 variable_manager=self.variable_manager,
                 loader=self.loader,
-                options=self.options,
+                options=self.cli.options,
                 passwords=None,
                 stdout_callback=callback,
             )

